onnxruntime/core/graph/graph.cc

```diff
@@ -614,6 +614,11 @@ void Node::ToProto(NodeProto& proto, bool update_subgraphs) const {
   if (!description_.empty())
     proto.set_doc_string(description_);
 
+  // Checks an attribute was not removed.
+  if (!can_be_saved_) {
```

Comment:
> if (!can_be_saved_)  [](http://example.com/codeflow?start=1&length=21)

I am expecting complaints from customers who would like to use optimized models. I think this should be made configurable.

Reply:
The removal happens after the optimized model is saved. This exception should never be raised unless the call to the function PruneRemovableAttributes is moved. That's mostly why I added this test.